tingbot/graphics.py

```diff
@@ -217,6 +217,77 @@ def text(self, string, xy=None, color='grey', align='center', font=None, font_si
 
         self.image(text_image, xy=xy, align=align, scale=1)
 
+    def oval(self, xy=None, size=(150,100), color='grey', align='center'):
+        """
+        Draws an oval.
+        
+        Args:
+            xy (tuple): The position (x, y) to draw the oval, as measured from the top-left.
+            size (tuple): The size (width, height) of the oval.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the oval relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        # if xy is not none
+        if xy is not None:
+            # and if it contains more than one value
+            if hasattr(xy, '__len__'):
+                # it can contain more than two values
+                 if len(xy) >= 2:
+                    # as long as the first two are valid numbers.
+                    if not(isinstance(xy[0], numbers.Real) and isinstance(xy[1], numbers.Real)):
+                        raise TypeError('xy should contain two numbers')
+        
+        # if size is not none
+        if size is not None:
+            # it has to contain at least two values:
+            if not(hasattr(size, '__len__')):
+                raise TypeError('size should contain at least two numbers')
+            # it can contain more than two values 
+            elif len(size) >= 2:
+                # as long as the first two are valid numbers
+                if not(isinstance(size[0], numbers.Real) and isinstance(size[0], numbers.Real)):
+                    raise TypeError('size should contain two numbers')
+                # and positive
+                elif size[0] < 0 or size[1] < 0:
+                    raise TypeError('the first two elements of size should be positive numbers')
+                # all checks passed, we need to extract the first two elements of size to draw the shape
+                else:
+                    xy = _topleft_from_aligned_xy(xy, align, (size[0:2]), self.size)
+                    pygame.draw.ellipse(self.surface, _color(color), pygame.Rect(xy, size[0:2]), 0) 
+            else:
+                xy = _topleft_from_aligned_xy(xy, align, size, self.size)
+                pygame.draw.ellipse(self.surface, _color(color), pygame.Rect(xy, size), 0) 
```

Comment:
instead of checking that size is greater than 2 and having the size[0:2] stuff, i'd just raise an error if the user passed in too many elements (there's basically no reason a user would do this on purpose). that simplifies this code a bit.